src/notation/view/abstractnotationpaintview.cpp

```diff
@@ -1284,10 +1292,13 @@ void AbstractNotationPaintView::movePlaybackCursor(midi::tick_t tick)
         return;
     }
 
-    if (configuration()->isAutomaticallyPanEnabled() && m_autoScrollEnabled) {
+    if (configuration()->isAutomaticallyPanEnabled()) {
         bool adjustVertically = needAdjustCanvasVerticallyWhilePlayback(newCursorRect);
 
-        if (adjustCanvasPosition(newCursorRect, adjustVertically)) {
+        if (configuration()->isSmoothPanning() && adjustCanvasPositionSmoothPan(newCursorRect)) {
+            return;
+        }
```

Comment:
The `adjustVertically` variable is only needed if the `if` condition is false. Since the method `needAdjustCanvasVerticallyWhilePlayback` is not quite trivial, and this code will be called very often, it would be preferable to call the method only if strictly necessary.